worker/lib/index.ts

```diff
@@ -280,6 +291,119 @@ async function setup(): Promise<express.Application> {
 		},
 	);
 
+	// give SSH keys to access DUT (is this the best way of doing this???)
+	app.post(
+		'/ssh/setup',
+		jsonParser,
+		async (
+			req: express.Request,
+			res: express.Response,
+			next: express.NextFunction,
+		) => {
+			try {
+				const keyPath = join(homedir(), 'id');
+				const keyPathPub = join(homedir(), 'id.pub');;
+				// need to write these strings to files
+				writeFileSync(keyPath, req.body.id);
+				writeFileSync(keyPath, req.body.id_pub);
+				await execSync('ssh-add -D');
+				await execSync(`ssh-add ${keyPath}`);
+				res.send('OK');
+			} catch (err) {
+				console.error(err);
+				next(err);
+			}
+		},
+	);
+
+	app.post(
+		'/dut/exec',
+		jsonParser,
+		async (
+			req: express.Request,
+			res: express.Response,
+			next: express.NextFunction,
+		) => {
+			try {
+				if (req.body.cmd != null){
+					if(req.body.target != null){
+						let cmd = req.body.cmd;
+						let target = await resolveLocalTarget(`${req.body.target}.local`);
+						// execute command over ssh here - TODO - do we keep the retries?
+						const result = await executeCommandOverSSH(
+							`source /etc/profile ; ${cmd}`,
+							{
+								host: target,
+								port: '22222',
+								username: 'root',
+							},
+						);
+		
+						if (typeof result.code === 'number' && result.code !== 0) {
+							throw new Error(
+								`"${cmd}" failed. stderr: ${result.stderr}, stdout: ${result.stdout}, code: ${result.code}`,
+							);
+						}
+		
+						res.send(result.stdout);
+					}
+						
+					} else {
+						throw new Error('Invalid request');
+					}
+			} catch (err) {
+				console.error(err);
+				next(err);
+			}
+		},
+	);
+
+	app.post(
+		`/exec`,
```

Comment:
This is to exec a command on the worker? Can we get some comments?

Reply:
@rcooke-warwick Maybe transfer the comments from the helpers.

Reply:
Thats right, this is to exec a command on the worker/testbot. I was 50/50 on whether we should allow people to that as it's un-contrained, but it's a way of getting the worker to do some things without more endpoints - e.g the stuff we do in the bluetooth test right now.